yksgi/src/lib.rs

```diff
@@ -149,6 +168,271 @@ impl SGInterp {
         }
     }
 
+    // Generate frames from stackmap information after a guard failure. The new frames are stored
+    // inside some allocated memory whose pointer this function returns. The frames are then later
+    // `memcpy`ed to the actual stack by ykcapi::__ykrt_reconstruct_frames.
+    pub fn reconstruct_frames(&self, btmframeaddr: *mut c_void) -> *const c_void {
+        // Load the stackmap and text sections from the binary. The former we need to parse in the
+        // stackmaps. The latter is required to disassemble the binary in order to distinguish
+        // branches from calls.
+        // FIXME: Don't use current_exe.
+        let pathb = env::current_exe().unwrap();
+        let file = fs::File::open(&pathb.as_path()).unwrap();
+        let exemmap = unsafe { memmap2::Mmap::map(&file).unwrap() };
+        let object = object::File::parse(&*exemmap).unwrap();
+        let sec = object.section_by_name(".llvm_stackmaps").unwrap();
+        let text = object.section_by_name(".text").unwrap();
+
+        // Create a capstone dissassembler.
+        // so we can figure out the size of the instruction we
+        // return too in order to adjust the return address in our reconstructed stackmap.
+        let cs = Capstone::new()
+        .x86()
+        .mode(arch::x86::ArchMode::Mode64)
+        .syntax(arch::x86::ArchSyntax::Att)
+        .detail(true)
+        .build()
+        .expect("Failed to create Capstone object");
+
+        // Vec holding currently active register values.
+        let mut registers = vec![0; 16];
+
+        // Parse the stackmap.
+        let slice = unsafe { slice::from_raw_parts(sec.address() as *mut u8, usize::try_from(sec.size()).unwrap()) };
+        let smentries = StackMapParser::get_entries(slice);
+
+        // The final size of the memory we need to allocate. Inititialised with space to store
+        // registers for register recovery.
+        let mut memsize: usize = 15 * 8;
```

Comment:
8 -> `sizeof::<*const ()>` or similar.

Reply:
Fixed in [3e270bc](https://github.com/ykjit/yk/pull/578/commits/3e270bceda6db7073f58c4fd90cea45ea6bcadab)